pkg/util/node/node.go

```diff
@@ -103,20 +103,10 @@ func GetNodeHostIPs(node *v1.Node) ([]net.IP, error) {
 	return nodeIPs, nil
 }
 
-// GetNodeHostIP returns the provided node's "primary" IP; see GetNodeHostIPs for more details
-func GetNodeHostIP(node *v1.Node) (net.IP, error) {
-	ips, err := GetNodeHostIPs(node)
-	if err != nil {
-		return nil, err
-	}
-	// GetNodeHostIPs always returns at least one IP if it didn't return an error
-	return ips[0], nil
-}
-
-// GetNodeIP returns an IP (as with GetNodeHostIP) for the node with the provided name.
+// GetNodeIPs returns an IP (as with GetNodeHostIPs) for the node with the provided name.
 // If required, it will wait for the node to be created.
-func GetNodeIP(client clientset.Interface, name string) net.IP {
-	var nodeIP net.IP
+func GetNodeIPs(client clientset.Interface, name string) []net.IP {
```

Comment:
FWIW it looks like `GetNodeIP` is only used by kube-proxy, and it's not obvious that any other part of k/k would have any need for it. (It's only useful for components that are running on a node but aren't kubelet and aren't running in a pod.)

So maybe `GetNodeIPs` should be in `cmd/kube-proxy` or `pkg/proxy/util` and you can submit a separate PR to just remove `GetNodeIP` later? (which saves us needing a sig-node approver on this PR...)

Reply:
Good idea. I'll create a PR to move GetNodeIP. Should I do the GetNodeIP -> GetNodeIPs in that PR? Or keep it as simple as possible since it must be approved by sig-node?

Reply:
Oh, sorry I mis-read. It wasn't a move, but to create a new GetNodeIPs. I'll get to it...